source/encoder.c

```diff
@@ -1043,22 +1037,40 @@ CBOREncoder_encode_datetime(CBOREncoderObject *self, PyObject *value)
 static PyObject *
 CBOREncoder_encode_date(CBOREncoderObject *self, PyObject *value)
 {
-    PyObject *datetime, *ret = NULL;
-
-    if (PyDate_Check(value)) {
-        datetime = PyDateTimeAPI->DateTime_FromDateAndTime(
-                PyDateTime_GET_YEAR(value),
-                PyDateTime_GET_MONTH(value),
-                PyDateTime_GET_DAY(value),
-                0, 0, 0, 0, self->tz,
-                PyDateTimeAPI->DateTimeType);
-        if (datetime) {
-            ret = CBOREncoder_encode_datetime(self, datetime);
-            Py_DECREF(datetime);
-            return ret;
+    // semantic type 100 or 1004
+
+    PyObject *tmp, *ret = NULL;
+    const char *buf;
+    Py_ssize_t length;
+    Py_INCREF(value);
```

Comment:
I'm actually not sure because this is my first time writing a python c-extension. Do you have a suggestion how it should be?

Reply:
Since `value` is passed as an argument, and you're not saving it anywhere, it should be treated as a borrowed reference. That is, you don't need to increase or decrease its reference count.

Reply:
Another point is that `Py_INCREF()` wll crash if `value` is `NULL`, so checking for a null value _after_ incrementing the refcount makes no sense.